src/atom_collection.rs

```diff
@@ -50,8 +59,10 @@ impl AtomCollection {
         name: &str,
         node: Node,
         delimiter: Option<&str>,
+        scope_id: Option<&str>,
     ) -> FormatterResult<()> {
         log::debug!("Resolving {name}");
+        let mut rng = rand::thread_rng();
```

Comment:
Should go outside `resolve_capture()`.

Reply:
And when you move it, I suppose it could become a simple counter.

Reply:
Well spotted! It removes the dependency to `rand` altogether